src/Spice86/UserControls/AddressColumn.cs

```diff
@@ -0,0 +1,77 @@
+﻿namespace Spice86.UserControls;
+
+using Avalonia;
+using Avalonia.Media.TextFormatting;
+
+using AvaloniaHex.Rendering;
+
+using Spice86.Shared.Utils;
+
+using System;
+
+internal class AddressColumn : Column {
+    private Size _minimumSize;
+
+    static AddressColumn() {
+        IsUppercaseProperty.Changed.AddClassHandler<HexColumn, bool>(OnIsUpperCaseChanged);
+    }
+
+    /// <inheritdoc />
+    public override Size MinimumSize => _minimumSize;
+
+    /// <summary>
+    /// Defines the <see cref="IsUppercase"/> property.
+    /// </summary>
+    public static readonly StyledProperty<bool> IsUppercaseProperty =
+        AvaloniaProperty.Register<HexColumn, bool>(nameof(IsUppercase), true);
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the hexadecimal digits should be rendered in uppercase or not.
+    /// </summary>
+    public bool IsUppercase {
+        get => GetValue(IsUppercaseProperty);
+        set => SetValue(IsUppercaseProperty, value);
+    }
+
+    private static void OnIsUpperCaseChanged(HexColumn arg1, AvaloniaPropertyChangedEventArgs<bool> arg2) {
+        arg1.HexView?.InvalidateVisualLines();
+    }
+
+    /// <inheritdoc />
+    public override void Measure() {
+        if (HexView is null) {
+            _minimumSize = default;
+        } else {
+            TextLine dummy = CreateTextLine("00000000 (0xFFFF:0xFFFF):")!;
+            _minimumSize = new Size(dummy.Width, dummy.Height);
+        }
+    }
+
+    /// <inheritdoc />
+    public override TextLine? CreateTextLine(VisualBytesLine line) {
+        if (HexView is null)
+            throw new InvalidOperationException();
+
+        ulong address = line.Range.Start.ByteIndex;
+        ushort segment = ConvertUtils.ToAbsoluteSegment((uint)address);
+        ushort offset = (ushort)(address - segment);
+        string text = IsUppercase
+            ? $"{address:X8} ({ConvertUtils.ToSegmentedAddressRepresentation(segment, offset)}):"
+            : $"{address:x8} ({ConvertUtils.ToSegmentedAddressRepresentation(segment, offset)}):";
+
+        return CreateTextLine(text);
+    }
+
+    private TextLine? CreateTextLine(string text) {
+        if (HexView is null)
+            return null;
+
+        GenericTextRunProperties properties = GetTextRunProperties();
+        return TextFormatter.Current.FormatLine(
+            new SimpleTextSource(text, properties),
+            0,
+            double.MaxValue,
```

Comment:
can you add parameter names to these numbers to make reviewing on the web easier?

Reply:
File removed.